contracts/extensions/DeltaNeutralBasisTradingStrategyExtension.sol

```diff
@@ -0,0 +1,1456 @@
+            return 0;
+        }
+
+        // `accountValue` is always positive. Do not use absolute value of basePositionValue in the below equation,
+        //  to keep the sign of CLR same as that of basePositionValue.
+        return _actionInfo.basePositionValue.preciseDiv(accountValue);
+    }
+
+    /**
+     * Calculate the new leverage ratio. The methodology reduces the size of each rebalance by weighting
+     * the current leverage ratio against the target leverage ratio by the recentering speed percentage. The lower the recentering speed, the slower
+     * the leverage token will move towards the target leverage each rebalance.
+     *
+     * return int256          New leverage ratio
+     */
+    function _calculateNewLeverageRatio(int256 _currentLeverageRatio) internal view returns(int256) {
+        // Convert int256 variables to uint256 prior to passing through methodology
+        uint256 currentLeverageRatioAbs = _currentLeverageRatio.abs();
+        uint256 targetLeverageRatioAbs = methodology.targetLeverageRatio.abs();
+        uint256 maxLeverageRatioAbs = methodology.maxLeverageRatio.abs();
+        uint256 minLeverageRatioAbs = methodology.minLeverageRatio.abs();
+
+        // CLRt+1 = max(MINLR, min(MAXLR, CLRt * (1 - RS) + TLR * RS))
+        // a: TLR * RS
+        // b: (1- RS) * CLRt
+        // c: (1- RS) * CLRt + TLR * RS
+        // d: min(MAXLR, CLRt * (1 - RS) + TLR * RS)
+        uint256 a = targetLeverageRatioAbs.preciseMul(methodology.recenteringSpeed);
+        uint256 b = PreciseUnitMath.preciseUnit().sub(methodology.recenteringSpeed).preciseMul(currentLeverageRatioAbs);
+        uint256 c = a.add(b);
+        uint256 d = Math.min(c, maxLeverageRatioAbs);
+        uint256 newLeverageRatio = Math.max(minLeverageRatioAbs, d);
+
+        return _currentLeverageRatio > 0 ? newLeverageRatio.toInt256() : newLeverageRatio.toInt256().neg();
+    }
+
+    /**
+     * Calculate total notional rebalance quantity and chunked rebalance quantity in base asset units for engaging the SetToken. Used in engage().
+     * Leverage ratio (for the base asset) is zero before engage. We open a new base asset position with size equals to ( (collateralBalance/2) * targetLeverageRatio / baseAssetPrice)
+     * to gain (targetLeverageRatio * collateralBalance/2) worth of exposure to the base asset.
+     * Note: We can't use `_calculateChunkRebalanceNotional` function because CLR is 0 during engage and it would lead to a divison by zero error.
+     *
+     * return int256          Chunked rebalance notional in base asset units
+     * return int256          Total rebalance notional in base asset units
+     */
+    function _calculateEngageRebalanceSize(
+        LeverageInfo memory _leverageInfo,
+        int256 _targetLeverageRatio
+    )
+        internal
+        view
+        returns (int256, int256)
+    {
+        int256 collateralAmount = collateralToken.balanceOf(address(strategy.setToken))
+            .div(2)
```

Comment:
Doesn't this limit us to at max having 2x leverage? If you were to pass in a higher amount of leverage then you would be trying to buy more ETH than we have USDC for.

Reply:
If say we wanted to execute this with 4x leverage and we have $100 of collateral we would trade $80 of USDC for ETH and deposit $20 collateral into Perp and sell $80 worth of ETH Perps.

Reply:
Fixed it in 779642f